src/main/java/bio/terra/service/dataset/flight/create/DatasetCreateFlight.java

```diff
@@ -24,13 +28,19 @@ public DatasetCreateFlight(FlightMap inputParameters, Object applicationContext)
         DataLocationService dataLocationService = (DataLocationService) appContext.getBean("dataLocationService");
         BigQueryPdao bigQueryPdao = (BigQueryPdao) appContext.getBean("bigQueryPdao");
         IamService iamClient = (IamService) appContext.getBean("iamService");
+        ApplicationConfiguration appConfig = (ApplicationConfiguration) appContext.getBean("applicationConfiguration");
 
         // get data from inputs that steps need
         AuthenticatedUserRequest userReq = inputParameters.get(
             JobMapKeys.AUTH_USER_INFO.getKeyName(), AuthenticatedUserRequest.class);
         DatasetRequestModel datasetRequest =
             inputParameters.get(JobMapKeys.REQUEST.getKeyName(), DatasetRequestModel.class);
 
+        addStep(new VerifyAuthorizationStep(
+            iamClient,
+            IamResourceType.DATAREPO,
+            appConfig.getResourceId(),
+            IamAction.CREATE_DATASET));
```

Comment:
👏 I'm a big fan of pushing this into a step instead of having it in the controller, in case I haven't mentioned this before.